custom_components/jcihitachi_tw/switch.py

```diff
@@ -176,3 +178,75 @@ def turn_off(self):
         _LOGGER.debug(f"Turn {self.name} off")
         self.put_queue(status_name="wind_swingable", status_str_value="disabled")
         self.update()
+
+class JciHitachiIonSwitchEntity(JciHitachiEntity, SwitchEntity):
+    def __init__(self, thing, coordinator):
+        super().__init__(thing, coordinator)
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        return f"{self._thing.name} Ion"
+
+    @property
+    def is_on(self):
+        """Indicate whether ion is on."""
+        status = self.hass.data[DOMAIN][UPDATED_DATA].get(self._thing.name, None)
+        if status:
+            if status.ion == "disabled":
+                return False
+            else:
+                return True
+        return None
+
+    @property
+    def unique_id(self):
+        return f"{self._thing.gateway_mac_address}_ion_switch"
+
+    def turn_on(self):
+        """Turn ion on."""
+        _LOGGER.debug(f"Turn {self.name} on")
+        self.put_queue(status_name="Ion", status_str_value="enabled")
+        self.update()
+    
+    def turn_off(self):
+        """Turn ion off."""
+        _LOGGER.debug(f"Turn {self.name} off")
+        self.put_queue(status_name="Ion", status_str_value="disabled")
+        self.update()
+
+class JciHitachiKeypadLockSwitchEntity(JciHitachiEntity, SwitchEntity):
+    def __init__(self, thing, coordinator):
+        super().__init__(thing, coordinator)
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        return f"{self._thing.name} Keypad Lock"
+
+    @property
+    def is_on(self):
+        """Indicate whether keypad lock is on."""
+        status = self.hass.data[DOMAIN][UPDATED_DATA].get(self._thing.name, None)
+        if status:
+            if status.keypad_lock == "disabled":
```

Comment:
Is `keypad_lock` a valid status?

Reply:
是的，實測後可控制。

![螢幕擷取畫面 2024-07-17 113622](https://github.com/user-attachments/assets/8fb0b25a-3d2b-41ae-9f32-cd2f2223fac4)

Reply:
It's a bit strange because `KeypadLock` doesn't have a legacy name called `keypad_lock`. So it shouldn't work.

https://github.com/qqaatw/LibJciHitachi/blob/3dc8951486768f230d35d35b1df392eab67c2045/JciHitachi/model.py#L1845-L1850

I wonder why it works though.

Reply:
因為我在LibJciHitachi 已經填上該變數的數值，故能取得該狀態數值，
也在該python module 丟PR 了。

https://github.com/qqaatw/LibJciHitachi/pull/21

謝謝。